core/iwasm/libraries/lib-wasi-threads/lib_wasi_threads_wrapper.c

```diff
@@ -31,16 +39,57 @@ typedef struct {
 static int32
 allocate_thread_id()
 {
-    static int32 thread_id = 0;
-
-    int32 id;
+    int32 id = -1;
 
     os_mutex_lock(&thread_id_lock);
-    id = thread_id++;
+    if (avail_tids.pos == 0) { // Resize stack and push new thread ids
+        uint32 old_size = avail_tids.size;
+        uint32 new_size = avail_tids.size * 2;
+        if (new_size / 2 != avail_tids.size) {
```

Comment:
[minor] you might just compare `old_size` with the `new_size` to avoid additional division.

Reply:
I used the division to make it similar to L54. I'm not sure a simple comparison would work for L54 (maybe when overflowing/wrapping the result can be bigger than the initial value?)

Reply:
yes, one way to detect integer overflow here is to check if `new_size < avail_tids.size`. It's not a major callout though.